pkg/instancetype/instancetype.go

```diff
@@ -136,58 +124,21 @@ func (m *methods) storeInstancetypeRevision(vm *virtv1.VirtualMachine) (*appsv1.
 	return storedRevision, nil
 }
 
-func CreatePreferenceControllerRevision(vm *virtv1.VirtualMachine, revisionName string, preferenceApiVersion string, preferenceSpec *instancetypev1alpha1.VirtualMachinePreferenceSpec) (*appsv1.ControllerRevision, error) {
-
-	preferenceSpecPatch, err := json.Marshal(*preferenceSpec)
-	if err != nil {
-		return nil, err
-	}
-
-	specRevision := instancetypev1alpha1.VirtualMachinePreferenceSpecRevision{
-		APIVersion: preferenceApiVersion,
-		Spec:       preferenceSpecPatch,
-	}
-
-	revisionPatch, err := json.Marshal(specRevision)
-	if err != nil {
-		return nil, err
-	}
-
-	return &appsv1.ControllerRevision{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:            revisionName,
-			Namespace:       vm.Namespace,
-			OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(vm, virtv1.VirtualMachineGroupVersionKind)},
-		},
-		Data: runtime.RawExtension{Raw: revisionPatch},
-	}, nil
-
-}
-
 func (m *methods) createPreferenceRevision(vm *virtv1.VirtualMachine) (*appsv1.ControllerRevision, error) {
-
 	switch strings.ToLower(vm.Spec.Preference.Kind) {
 	case apiinstancetype.SingularPreferenceResourceName, apiinstancetype.PluralPreferenceResourceName:
```

Comment:
I forget why I did this but it shouldn't be possible to have a plural kind here right?

Reply:
Again, might be something to cleanup outside of this series but I don't think we can end up with a plural kind here.

Reply:
Most of other kubernetes tools also accept plural when expecting Kind. So I thought it would be good to also accept it here, so user is not surprised.

Reply:
ACK.